hal/include/bmp180_device.h

```diff
@@ -0,0 +1,179 @@
+	static const uint8_t k2MSDelay = 2;
+	static const uint8_t k3MSDelay = 3;
+	static const uint8_t kAverage = 3;
+	static const uint8_t kCheckDivisor = 0;
+	static const uint8_t kDataMeasure = 3;
+	static const uint8_t kCalculateTruePressure = 8;
+	static const uint8_t kCalculateTrueTemperature = 8;
+
+	enum {
+		kSHIFT_BIT_POSITION_BY_01_BITS = 1,
+		kSHIFT_BIT_POSITION_BY_02_BITS = 2,
+		kSHIFT_BIT_POSITION_BY_04_BITS = 4,
+		kSHIFT_BIT_POSITION_BY_06_BITS = 6,
+		kSHIFT_BIT_POSITION_BY_08_BITS = 8,
+		kSHIFT_BIT_POSITION_BY_11_BITS = 11,
+		kSHIFT_BIT_POSITION_BY_12_BITS = 12,
+		kSHIFT_BIT_POSITION_BY_13_BITS = 13,
+		kSHIFT_BIT_POSITION_BY_15_BITS = 15,
+		kSHIFT_BIT_POSITION_BY_16_BITS = 16,
+	};
+
+	// register definitions
+	static const int8_t kPromStartAddr = 0xAA;
+	static const int8_t kPromDataLen = 22;
+
+	static const int8_t kChipIDReg = 0xD0;
+	static const int8_t kVersionReg = 0xD1;
+
+	static const int8_t kCtrlMeasReg = 0xF4;
+	static const int8_t kADCOutMSBReg = 0xF6;
+	static const int8_t kADCOutLSBReg = 0xF7;
+
+	static const int8_t kSoftResetReg = 0xE0;
+
+	// temperature measurement
+	static const int8_t kTMeasure = 0x2E;
+	// pressure measurement
+	static const int8_t  kPMeasure = 0x34;
+
+	// TO be spec'd by GL or SB
+	static const uint8_t  kTempConversionTime = 5;
+
+	static const int16_t kParamMG = 3038;
+	static const int16_t kParamMH = -7357;
+	static const int16_t kParamMI = 3791;
+
+	// ARRAY SIZE DEFINITIONS
+	static const uint8_t kTemperatureDataBytes = 2;
+	enum {
+		kTEMPERATURE_MSB_DATA,
+		kTEMPERATURE_LSB_DATA
+	};
+
+	static const uint8_t kPressureDataBytes = 3;
+	enum {
+		kPRESSURE_MSB_DATA,
+		kPRESSURE_LSB_DATA,
+		kPRESSURE_XLSB_DATA,
+	};
+
+	static const uint8_t kCalibDataSize = 22;
+	enum {
+		kCALIB_PARAM_AC1_MSB,
+		kCALIB_PARAM_AC1_LSB,
+		kCALIB_PARAM_AC2_MSB,
+		kCALIB_PARAM_AC2_LSB,
+		kCALIB_PARAM_AC3_MSB,
+		kCALIB_PARAM_AC3_LSB,
+		kCALIB_PARAM_AC4_MSB,
+		kCALIB_PARAM_AC4_LSB,
+		kCALIB_PARAM_AC5_MSB,
+		kCALIB_PARAM_AC5_LSB,
+		kCALIB_PARAM_AC6_MSB,
+		kCALIB_PARAM_AC6_LSB,
+		kCALIB_PARAM_B1_MSB,
+		kCALIB_PARAM_B1_LSB,
+		kCALIB_PARAM_B2_MSB,
+		kCALIB_PARAM_B2_LSB,
+		kCALIB_PARAM_MB_MSB,
+		kCALIB_PARAM_MB_LSB,
+		kCALIB_PARAM_MC_MSB,
+		kCALIB_PARAM_MC_LSB,
+		kCALIB_PARAM_MD_MSB,
+		kCALIB_PARAM_MD_LSB,
+	};
+
+	// brief This structure holds all device specific calibration parameters
+	struct calib_param_t
+	{
+		int ac1; 			// <calibration ac1 data
+		int ac2; 			// <calibration ac2 data
+		int ac3; 			// <calibration ac3 data
+		unsigned int ac4; 	// <calibration ac4 data
+		unsigned int ac5; 	// <calibration ac5 data
+		unsigned int ac6; 	// <calibration ac6 data
+		int b1; 			// <calibration b1 data
+		int b2; 			// <calibration b2 data
+		int mb; 			// <calibration mb data
+		int mc; 			// <calibration mc data
+		int md; 			// <calibration md data/
+	};
+
+	struct calib_param_t calib_param_; /**<calibration data*/
+		uint8_t mode_; 				// <power mode
```

Comment:
Why the members in this struct are using something like `uint8_t` but the previous `struct calib_param_t` is using `int`?

Reply:
Because they are not members of the struct calib_param_t